### PR TITLE
tests(node-builder): basic exex test update for apply function

### PR DESCRIPTION
**Pull request by kien6034**
Tests for this PR [1162](https://github.com/paradigmxyz/reth/pull/11682)

### CODE REVIEW

File: crates/node/builder/src/builder/mod.rs [hunk 1]
```diff
@@ -177,6 +177,12 @@ impl<DB, ChainSpec> NodeBuilder<DB, ChainSpec> {
     pub const fn config(&self) -> &NodeConfig<ChainSpec> {
         &self.config
     }
+
+    /// Returns a mutable reference to the node builder's config.
+    #[cfg(feature = "test-utils")]
```
Comment (mattsse) [inline crates/node/builder/src/builder/mod.rs:182]: we can actually make this non feature gated

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/node/builder/src/builder/mod.rs (hunk 1 above)
Comment (kien6034) [reply crates/node/builder/src/builder/mod.rs]: yeah i updated

**Review by mattsse — APPROVED** [review]

File: crates/ethereum/node/tests/it/exex.rs [hunk 2]
```diff
@@ -30,6 +30,10 @@ fn basic_exex() {
     let config = NodeConfig::test();
     let db = create_test_rw_db();
     let _builder = NodeBuilder::new(config)
+        .apply(|mut builder| {
+            builder.config_mut().dev.dev = true;
```
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/ethereum/node/tests/it/exex.rs]: yeah although this entire function only does typechecking, be we can add the function as is

[the remark this answers is not in the document]
Comment (kien6034) [reply crates/ethereum/node/tests/it/exex.rs]: i assume i should write test in other file / function?

**Comment by shekhirin** [comment]
@mattsse the PR title is now confusing 😅

**Pull request merged**